### PR TITLE
treewide: "KDE6" prep, part 1

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

#266868, but only the stable, mergeable bits

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
with this LGTM

File: pkgs/development/libraries/mlt/default.nix [hunk 1]
```diff
@@ -31,6 +31,7 @@
 , swig
 , enableQt ? false
 , libsForQt5
```
Comment (peterhoeg) [inline pkgs/development/libraries/mlt/default.nix:33]: Redundant given the next line?

Comment (K900) [reply pkgs/development/libraries/mlt/default.nix:33]: Technically we could make it `qt = null`, yeah.

**Comment by K900** [comment]
Applied suggestions, retargeted to staging because rebuilds.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/qca/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchurl, cmake, openssl, pkg-config, qtbase, qt5compat ? null }:
+
+let
+  isQt6 = lib.versions.major qtbase.version == "6";
+in stdenv.mkDerivation rec {
+  pname = "qca";
+  version = "2.3.7";
+
+  src = fetchurl {
+    url = "http://download.kde.org/stable/qca/${version}/qca-${version}.tar.xz";
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/libraries/qca/default.nix:10]: did all the others in #271540 to hopefully not create merge conflicts with this one

**Review by SuperSandro2000 — APPROVED** [review]

**Review by philiptaron — APPROVED** [review]
Everything _looks_ good; let's let the machines have their way and say what else is broken. One minor suggestion about using a commit hash instead of `main`.

**Review by philiptaron — APPROVED** [review]

**Comment by K900** [comment]
AppStream breaks like all of Pantheon, going to merge without it.

**Comment by bobby285271** [comment]
> AppStream breaks like all of Pantheon

:ok_hand: :upside_down_face: 

I tried building `nixosTests.pantheon` on https://github.com/K900/nixpkgs/commit/4c0948c0246acae4f7e744c0d69f3ae1575189d9, so far the 2 failures for me are:

- pantheon.switchboard-plug-about: fix in https://github.com/NixOS/nixpkgs/pull/272049
- pantheon.appcenter: assume we do need to land appstream 1.0 I am fine to point src.rev to https://github.com/elementary/appcenter/commit/d93e135a0b0c9a6e0fbad18fe90d46425823a42c

**Comment by K900** [comment]
There's no rush really, Plasma 6 stable is in February, so hopefully things switch over by that time.

**Pull request merged**